src/main/java/uk/gov/ons/ctp/response/action/service/impl/ActionRuleServiceImpl.java

```diff
@@ -43,4 +43,45 @@ public ActionRule createActionRule(final ActionRule actionRule) {
     actionRule.setId(UUID.randomUUID());
     return actionRuleRepo.saveAndFlush(actionRule);
   }
+
+  @Override
+  @Transactional(propagation = Propagation.REQUIRED, timeout = TRANSACTION_TIMEOUT)
+  public ActionRule updateActionRule(final ActionRule actionRule) {
+    final UUID actionRuleId = actionRule.getId();
+
+    ActionRule existingActionRule = actionRuleRepo.findById(actionRuleId);
+    if (existingActionRule != null) {
+      boolean needsUpdate = false;
+
+      final Integer newPriority = actionRule.getPriority();
+      if (newPriority != null) {
+        needsUpdate = true;
+        existingActionRule.setPriority(newPriority);
```

Comment:
`existingActionRule` is a bit of a misleading name here since we're manipulating it into a new ActionRule
Could we map the new attributes onto a new ActionRule object so we don't have to reassign?